--- a/mbedtls_transport.c	2024-12-11 11:35:41.003489100 +0100
+++ b/mbedtls_transport.c	2024-12-11 11:35:33.982165100 +0100
@@ -463,7 +463,7 @@
 {
     TLSContext_t * pxTLSCtx = NULL;
 
-    pxTLSCtx = ( TLSContext_t * ) pvPortMalloc( sizeof( TLSContext_t ) );
+    pxTLSCtx = ( TLSContext_t * ) mbedtls_platform_calloc( 1, sizeof( TLSContext_t ) );
 
     if( pxTLSCtx == NULL )
     {
@@ -1473,7 +1473,7 @@
 
     if( pxNotifyThreadCtx == NULL )
     {
-        pxNotifyThreadCtx = pvPortMalloc( sizeof( NotifyThreadCtx_t ) );
+        pxNotifyThreadCtx = mbedtls_platform_calloc(1, sizeof( NotifyThreadCtx_t ) );
 
         if( pxNotifyThreadCtx == NULL )
         {
@@ -1791,4 +1791,4 @@
 
     vLoggingPrintf( pcLogLevel, pcFileBaseName, lLineNumber, pcErrStr );
 }
-#endif /* ifdef MBEDTLS_DEBUG_C */
\ No newline at end of file
+#endif /* ifdef MBEDTLS_DEBUG_C */
